pkg/blob/controllerserver.go

```diff
@@ -366,36 +366,33 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 	defer d.volumeLocks.Release(volName)
 
 	requestName := "controller_create_volume"
-
-	var srcAzcopyAuthEnv []string
-	var srcSubscriptionID, srcResourceGroupName, srcAccountName, srcContainerName, srcPath, srcAccountSASToken string
 	if volContentSource != nil {
 		switch volContentSource.Type.(type) {
 		case *csi.VolumeContentSource_Snapshot:
-			return nil, status.Errorf(codes.InvalidArgument, "VolumeContentSource Snapshot is not yet implemented")
+			requestName = "controller_create_volume_from_snapshot"
 		case *csi.VolumeContentSource_Volume:
 			requestName = "controller_create_volume_from_volume"
-			var srcVolumeID string
-			if volContentSource.GetVolume() != nil {
-				srcVolumeID = volContentSource.GetVolume().GetVolumeId()
-			}
-			srcResourceGroupName, srcAccountName, srcContainerName, _, srcSubscriptionID, err = GetContainerInfo(srcVolumeID)
-			if err != nil {
-				return nil, status.Error(codes.NotFound, err.Error())
-			}
-			srcAccountOptions := &azure.AccountOptions{
-				Name:                srcAccountName,
-				SubscriptionID:      srcSubscriptionID,
-				ResourceGroup:       srcResourceGroupName,
-				GetLatestAccountKey: getLatestAccountKey,
-			}
-			srcAccountSASToken, srcAzcopyAuthEnv, err = d.getAzcopyAuth(ctx, srcAccountName, "", storageEndpointSuffix, srcAccountOptions, secrets, secretName, secretNamespace)
-			if err != nil {
-				return nil, status.Errorf(codes.Internal, "failed to getAzcopyAuth on account(%s) rg(%s), error: %v", accountOptions.Name, accountOptions.ResourceGroup, err)
-			}
-			srcPath = fmt.Sprintf("https://%s.blob.%s/%s", srcAccountName, storageEndpointSuffix, srcContainerName)
-		default:
-			return nil, status.Errorf(codes.InvalidArgument, "VolumeContentSource is not recognized: %v", volContentSource)
+			// 	var srcVolumeID string
```

Comment:
pls remove commented code if it's not needed any more

Reply:
done for mistake